src/main/java/picard/sam/DownsampleSam.java

```diff
@@ -212,7 +220,45 @@ protected int doWork() {
         }
 
         final SamReader in = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(SamInputResource.of(INPUT));
-        final SAMFileWriter out = new SAMFileWriterFactory().makeSAMOrBAMWriter(in.getFileHeader(), true, OUTPUT);
+        final SAMFileHeader header = in.getFileHeader().clone();
+
+        if (STRATEGY == Strategy.ConstantMemory || STRATEGY == Strategy.Chained) {
+            //if running using ConstantMemory or Chained strategy, need to check if we have previously run with the same random seed
+            //collect previously used seeds
+            final int USER_SEED = RANDOM_SEED;
+            final Set<Integer> previousSeeds = new HashSet<>();
+            for (final SAMProgramRecord pg : header.getProgramRecords()) {
+                if (pg.getProgramName() != null && pg.getProgramName().equals(PG_PROGRAM_NAME)) {
+                    final String previousSeedString = pg.getAttribute(RANDOM_SEED_TAG);
+                    if (previousSeedString == null) {
+                        /* The previous seed was not recorded.  In this case, the current seed may be the same as the previous seed,
+                        so we will change it to a randomly selected seed, which is very likely to be unique
+                         */
+                        RANDOM_SEED = new Random(pg.hashCode()).nextInt();
+                        log.warn("DownsampleSam has been run before on this data, but the previous seed was not recorded.  The used seed will be changed to minimize the chance of using the" +
+                                " same seed as in a previous run.");
+                    }
+                    final int previousSeed = Integer.parseInt(previousSeedString);
+                    previousSeeds.add(previousSeed);
+                }
+            }
+
+            final Random rnd = new Random(RANDOM_SEED);
+            while (previousSeeds.contains(RANDOM_SEED)) {
+                final int previousSeed = RANDOM_SEED;
+                RANDOM_SEED = rnd.nextInt();
+                log.warn("DownsampleSam has been run before on this data with the seed .  The random seed will be changed to avoid using the " +
```

Comment:
extra space before period...were you going to put a more informative error message here?